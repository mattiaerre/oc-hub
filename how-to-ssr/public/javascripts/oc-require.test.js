const wrapper = require('./oc-require');

describe('oc-require', () => {
  const head = { load: jest.fn((url, callback) => { callback(); }) };
  const $window = { we: { love: 'oc' } };
  const url = 'https://mock.url/dep.min.js';
  const callback = jest.fn();

  beforeEach(() => {
    callback.mockReset();
  });

  test('when dep doesn\'t need to be loaded, callback receives dep', async () => {
    const method = wrapper(undefined, $window);

    const nameSpace = 'we';
    await method(nameSpace, undefined, callback);

    expect(callback).toHaveBeenCalled();
    expect(callback.mock.calls).toMatchSnapshot();
  });

  test('when nameSpace undefined, callback receives undefined', async () => {
    const method = wrapper(head, undefined);

    await method(url, callback);

    expect(callback).toHaveBeenCalled();
    expect(callback.mock.calls).toMatchSnapshot();
  });

  test('when nameSpace not undefined and matches, callback receives dep', async () => {
    const method = wrapper(head, $window);

    await method('we', url, callback);

    expect(callback).toHaveBeenCalled();
    expect(callback.mock.calls).toMatchSnapshot();
  });

  test('when nameSpace not undefined but doesn\'t match, callback receives undefined', async () => {
    const method = wrapper(head, $window);

    await method('you', url, callback);

    expect(callback).toHaveBeenCalled();
    expect(callback.mock.calls).toMatchSnapshot();
  });
});
