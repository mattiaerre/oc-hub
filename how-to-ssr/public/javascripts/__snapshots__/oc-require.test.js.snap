// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`oc-require when dep doesn't need to be loaded, callback receives dep 1`] = `
Array [
  Array [
    Object {
      "love": "oc",
    },
  ],
]
`;

exports[`oc-require when nameSpace not undefined and matches, callback receives dep 1`] = `
Array [
  Array [
    Object {
      "love": "oc",
    },
  ],
]
`;

exports[`oc-require when nameSpace not undefined but doesn't match, callback receives undefined 1`] = `
Array [
  Array [
    undefined,
  ],
]
`;

exports[`oc-require when nameSpace undefined, callback receives undefined 1`] = `
Array [
  Array [
    undefined,
  ],
]
`;
